File: php/testcode/securecookie_042.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cookie_random_bytes_value
function securecookie042(BenchmarkRequest $req): BenchmarkResponse {
    $token = bin2hex(random_bytes(32));
    setcookie('session', $token, ['secure' => true, 'httponly' => true, 'samesite' => 'Lax']); // vuln-code-snippet safe-line php_cookie_random_bytes_value
    return BenchmarkResponse::ok('cookie set');
}
// vuln-code-snippet end php_cookie_random_bytes_value

File: ruby/testcode/securecookie_042.rb
require_relative 'shared'
require 'securerandom'

# vuln-code-snippet start ruby_securecookie_hsts_pair
def set_session_with_hsts(req)
  token = SecureRandom.hex(32)
  response = BenchmarkResponse.ok('session set')
  response.headers['Strict-Transport-Security'] = 'max-age=31536000; includeSubDomains'
  response.headers['Set-Cookie'] = "session=#{token}; Secure; HttpOnly; SameSite=Strict; Path=/" # vuln-code-snippet safe-line ruby_securecookie_hsts_pair
  response
end
# vuln-code-snippet end ruby_securecookie_hsts_pair
